misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlLexer.java

```diff
@@ -672,4 +786,27 @@ private int peek() {
     private String getLexeme() {
         return reader.getMarkedChars();
     }
+
+    /**
+     * Process string numeric escape.
+     * <p>
+     * <code>StringNumericEscape := \u00E9 </code>
+     */
+    private void processStringNumericEscape() {
+        // Process '\ u {'
+        this.reader.advance(3);
```

Comment:
you reach here after peeking only 2 tokens. So this should advance only 2 tokens.
Comment in L#796 is also not correct.